tests/introspection/tests.py

```diff
@@ -189,6 +184,15 @@ def test_get_indexes_multicol(self):
         self.assertNotIn('first_name', indexes)
         self.assertIn('id', indexes)
 
+    @skipUnlessDBFeature('can_introspect_index_type')
+    def test_get_constraints_indexes_types(self):
+        with connection.cursor() as cursor:
+            constraints = connection.introspection.get_constraints(cursor, Article._meta.db_table)
+        for key, val in constraints.items():
+            if val['index'] and not (val['primary_key'] or val['unique']):
```

Comment:
I have a put a counter to the loop instead.
The thing is that, that line will be changed to `index_type = indexes['table_name_column_k3Ha1Q_idx'].pop('type', None)` and I don't want think it would be good idea to make this test dependent on a name auto-generated by Django.